setup.py

```diff
@@ -221,7 +223,25 @@ def get_extensions():
         except ImportError:
             pass
 
-        if is_rocm_pytorch or torch.cuda.is_available() or os.getenv(
+        if os.getenv('MMCV_WITH_DIOPI', '0') == '1':
+            print(f'Compiling {ext_name} with CPU and DIPU')
+            define_macros += [('MMCV_WITH_DIOPI', None)]
+            define_macros += [('DIOPI_ATTR_WEAK', None)]
+            op_files = glob.glob('./mmcv/ops/csrc/pytorch/*.cpp') + \
+                glob.glob('./mmcv/ops/csrc/pytorch/cpu/*.cpp')
+            extension = CppExtension
+            include_dirs.append(os.path.abspath('./mmcv/ops/csrc/common'))
+            dipu_root = os.getenv('DIPU_ROOT')
```

Comment:
Is all of these env (`DIPU_ROOT`, `DIOPI_PATH`, `DIPU_PATH`, `VENDOR_INCLUDE_DIRS`) required? If some of it are optional, assign a default value would be cool.

Reply:
yes, these are required. the meaning of its should alwayd corrsponds to ```dipu(torch_dipu)``` .which  is under rapid devlopment.(https://github.com/DeepLink-org/dipu_poc) . hopefully, these env para would be provided by ```dipu(torch_dipu)``` in the final. Regard ```dipu(torch_dipu)``` as one pytorch ext which will support a general device called ```dipu```.

Reply:
would you mind providing me the access to the repo?

Reply:
invitaion sent